var/spack/repos/builtin/packages/libfabric/package.py

```diff
@@ -44,6 +48,7 @@ class Libfabric(AutotoolsPackage):
     version("1.4.2", sha256="5d027d7e4e34cb62508803e51d6bd2f477932ad68948996429df2bfff37ca2a5")
 
     fabrics = (
+        "cxi",
```

Comment:
FWIW, I'm not in favor of adding this as a fabric option here as the cxi provide is still not available upstream, where we pull the releases from. See discussion in https://github.com/spack/spack/pull/28446

I am not a spack expert, but this could be handled differently with a site-specific config and some combination of dynamically loading the external package and setting FI_PROVIDER=cxi at runtime??

Reply:
@rajachan I understand that the `cxi` code is not available upstream, I believe that we have had spack packages have variants and features for non-publicly accessible code bases.  I think that `mvapich2-gdr` and `cray-libsci` are an examples of such packages.  The key thing that I think that we would want to recognize is that CXI is a valid variant for some systems.  @becker33 @tgamblin How should we proceed?

Reply:
cc: @shefty for visibility

Reply:
I'm unfamiliar with spack and the intent here.  But libfabric can make use of external provider libraries.  If there's some way to install a cxi package separately, and all this is doing is letting libfabric know that it might exists, this makes sense.  If this is assuming cxi is part of the libfabric package, then I agree adding it here may not make sense.  (Basically, I don't know. :)

Reply:
A variant is the equivalent of toggling --enable-<provider> so this is a build time thing. I also think this should use the provider as if it was an external provider, discovering at runtime.

Reply:
@rajachan and @becker33  Okay, I understand the issue here.  I have tried to mitigate it by making the CXI fabric only available when on a cray based system that has a slingshot fabric.  This still doesn't address the issue that the code isn't available for public builds, but should avoid impacting other instances of the package.  Thoughts?  Also it needs to be rebased on #33796 and #33939.

Reply:
Yes, this looks a lot better! 👍